### DISCUSSION TITLE
Suddenly 2 paypal folders.

### DISCUSSION THREAD

**Discussion by HansF** [pretix/pretix#3601] [Q&A] [answered]
Hey, I guess I did something wrong, but somehow I've ended up with 2 paypal folders in my plugins folder. 

> /var/pretix/venv/lib/python3.10/site-packages/pretix/plugins# diff -q paypal/ paypal2/
> Files paypal/__init__.py and paypal2/__init__.py differ
> Common subdirectories: paypal/__pycache__ and paypal2/__pycache__
> Only in paypal/: api.py
> Files paypal/apps.py and paypal2/apps.py differ
> Only in paypal2/: client
> Common subdirectories: paypal/migrations and paypal2/migrations
> Only in paypal/: models.py
> Files paypal/payment.py and paypal2/payment.py differ
> Files paypal/signals.py and paypal2/signals.py differ
> Common subdirectories: paypal/static and paypal2/static
> Common subdirectories: paypal/templates and paypal2/templates
> Files paypal/urls.py and paypal2/urls.py differ
> Files paypal/views.py and paypal2/views.py differ

I only noticed this after the install of 3.7.1, don't know if it was there before. I noticed it because there are now two paypal plugins at the update page. 

![image](https://github.com/pretix/pretix/assets/1503/a8c8154c-fb38-4d84-8f90-30c14a04c640)

I don't use the paypal plugin at the moment. So I didn't check how it performs. 
How can I resolve this? Any ideas? 
Can this be caused be me forgetting to run migrate before rebuild and updatestyles and running it after?

**Comment by pc-coholic** [comment, marked as answer]
Hi!

Nope, everything is alright ;-)

We rewrote the whole PayPal integration, added some magic glue code to switch automatically to the new one whenever possible and shopped it :)

Yes, you will see two PayPal plugins on the update page and have two PayPal-folders - but that is intended.

The changed happened with prefix 4.11.0: https://pretix.eu/about/en/blog/20220701-release-4110/

**Comment by HansF** [comment]
Super, thanks for letting me know so quickly. Have a nice weekend!
